### PR TITLE
fix(coachmark): add "step-count" slot for custom/internationalized pagination content

### PR DESCRIPTION
**Pull request by Rajdeepc**
## Description

This PR adds the ability for users to customize the separator between the current step and the total steps in the YourComponent component. By default, the separator is 'of', but users can now specify a custom separator using the separator attribute.

### Changes Made
Added a new separator property to the YourComponent component to allow customization of the separator.
Updated the component's template to use the separator property for the separator between the current step and the total steps.

## Related issue(s)

- #4213 

## How has this been tested?

-   [x] _Test case 1_
    1. Go [here](https://feature-coachmark-localization--spectrum-web-components.netlify.app/storybook/?path=/story/coachmark--custom-separator)
    2. Use the separator attribute to specify a custom separator.
    3. Verify that the component renders with the custom separator between the current step and the total steps.
-   [ ] _Test case 2_
    1. Go here
    2. Do this

## Screenshots (if appropriate)
Screenshot 2024-03-26 at 4 23 54 PM

## Types of changes

-   [ ] Bug fix (non-breaking change which fixes an issue)
-   [x] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to change)
-   [ ] Chore (minor updates related to the tooling or maintenance of the repository, does not impact compiled assets)

## Checklist

-   [x] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
-   [x] My code follows the code style of this project.
-   [x] If my change required a change to the documentation, I have updated the documentation in this pull request.
-   [x] I have read the **[CONTRIBUTING](<(https://github.com/adobe/spectrum-web-components/blob/main/CONTRIBUTING.md)>)** document.
-   [x] I have added tests to cover my changes.
-   [x] All new and existing tests passed.
-   [x] I have reviewed at the Accessibility Practices for this feature, see: [Aria Practices](https://www.w3.org/TR/wai-aria-practices/)

## Best practices

This repository uses conventional commit syntax for each commit message; note that the GitHub UI does not use this by default so be cautious when accepting suggested changes. Avoid the "Update branch" button on the pull request and opt instead for rebasing your branch against `main`.

### CODE REVIEW

**Comment by Westbrook** [comment]
Can you sync with the internationalization team to confirm that this works across the actual usage of our target languages? Naively, I've seeing `1 of 4` translate to `4 件中 1 件` in Japanese which I'm not sure how this API covers. Even a greatly simplified `４分の１` which might benefit from this change requires a content order change that I'm not sure a good path to managing here.

**Comment by Rajdeepc** [comment]
> Can you sync with the internationalization team to confirm that this works across the actual usage of our target languages? Naively, I've seeing `1 of 4` translate to `4 件中 1 件` in Japanese which I'm not sure how this API covers. Even a greatly simplified `４分の１` which might benefit from this change requires a content order change that I'm not sure a good path to managing here.

I had a talk with the Internationalisation team and they said we cannot do string concatenation. We have to provide them a slot so that the entire string should be populated by the user.

File: packages/coachmark/src/Coachmark.ts [hunk 1]
```diff
@@ -273,7 +273,7 @@ export class Coachmark extends Popover {
         return html`
             <div class="step" role="status">
                 <span aria-live="polite">
-                    ${this.currentStep} of ${this.totalSteps}
+                    <slot name="step-count"></slot>
```
Comment (Westbrook) [inline packages/coachmark/src/Coachmark.ts:276]:
```suggestion
                    <slot name="step-count">${this.currentStep} of ${this.totalSteps}</slot>
```
I think we should default to the English delivery here in order to reduce chances of happy path issues.

File: packages/coachmark/README.md [hunk 2]
```diff
@@ -66,6 +66,35 @@ Coach marks can include an `<sp-action-menu>`, which appears at the top right of
...
+Coach mark in tour by default, uses the separator `'of'` between the current step and the total steps. You can customize this separator by using the `separator` attribute. If no `separator` is provided, it will default to `'of'`.
+
+```html
+<sp-coachmark
+    open
+    current-step="2"
+    total-steps="8"
+    primary-cta="Next"
+    secondary-cta="Previous"
+    separator="out of"
```
Comment (Westbrook) [inline packages/coachmark/README.md:80]: We're removed this property, correct?

**Review by Westbrook — COMMENTED** [review]
I'd like one of the stories to deliver a previously unachievable page count text to ensure that we can continue to achieve it over time.

**Comment by Rajdeepc** [comment]
@Westbrook Let me know if we are good to close this or do you see any additional work that needs to be done here?

**Comment by Westbrook** [comment]
> I'd like one of the stories to deliver a previously unachievable page count text to ensure that we can continue to achieve it over time.

Can we see this in a story that surfaces to VRT to ensure we're covered over time? Something like `４分の１` or other.

**Review by Westbrook — APPROVED** [review]
LGTM! :shipit: 🎆

**Review by Westbrook — CHANGES_REQUESTED** [review]
Sorry, I missed one thing that I can't place inline...

Can you add:
```
 * @slot step-count - Override the default pagination delivery with your own internationalized content
```
At or around line 40 so that it gets into the docs site.

**Review by Westbrook — APPROVED** [review]
:shipit: but, like, for real this time 😉

**Pull request merged**